lib/models/addon.js

```diff
@@ -1097,17 +1093,48 @@ let addonProto = {
   jshintAddonTree() {
     this._requireBuildPackages();
 
+    let lintTrees = [];
+
     let addonPath = this._treePathFor('addon');
+    if (existsSync(addonPath)) {
+      let addonJs = this.addonJsFiles(addonPath);
+      let lintAddonJsTrees = this._eachProjectAddonInvoke('lintTree', ['addon', addonJs]);
+
+      let addonTemplates = this._addonTemplateFiles(addonPath);
+      let lintTemplateTrees = this._eachProjectAddonInvoke('lintTree', ['templates', addonTemplates]);
+
+      lintTrees = lintTrees.concat(lintAddonJsTrees, lintTemplateTrees);
+    }
+
+    let appPath = this._treePathFor('app');
+    if (existsSync(appPath)) {
+      let appJs = this.appJsFiles(appPath);
```

Comment:
Ya, most of them are `broccoli-persistent-filter`'s with `extensions = ['js']` so it wouldn't really matter...